src/lib/viewers/controls/page/__tests__/PageControlsForm-test.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { shallow, ShallowWrapper } from 'enzyme';
+import noop from 'lodash/noop';
+import PageControlsForm, { ENTER, ESCAPE } from '../PageControlsForm';
+
+describe('PageControlsForm', () => {
+    const getWrapper = (props = {}): ShallowWrapper =>
+        shallow(<PageControlsForm onPageSubmit={noop} pageCount={3} pageNumber={1} {...props} />);
+    const getFormButton = (wrapper: ShallowWrapper): ShallowWrapper =>
+        wrapper.find('[data-testid="bp-PageControlsForm-button"]');
+    const getFormInput = (wrapper: ShallowWrapper): ShallowWrapper =>
+        wrapper.find('[data-testid="bp-PageControlsForm-input"]');
+
+    describe('event handlers', () => {
+        test.each`
+            newPageNumber | onPageSubmitCallCount
+            ${2}          | ${1}
+            ${''}         | ${0}
+        `('should call onPageSubmit when input blurs', ({ newPageNumber, onPageSubmitCallCount }) => {
+            const onPageSubmit = jest.fn();
+            const pageCount = 3;
+            const pageNumber = 1;
+            const wrapper = getWrapper({ onPageSubmit, pageCount, pageNumber });
+
+            getFormButton(wrapper).simulate('click');
+            getFormInput(wrapper).simulate('change', { target: { value: newPageNumber } });
+            getFormInput(wrapper).simulate('blur');
+
+            expect(onPageSubmit.mock.calls.length).toBe(onPageSubmitCallCount);
+        });
+
+        test.each`
+            newPageNumber | onPageSubmitCallCount
+            ${''}         | ${0}
+            ${2}          | ${1}
+        `('should handle when Enter key is pressed on input', ({ onPageSubmitCallCount, newPageNumber }) => {
+            const key = ENTER;
```

Comment:
Some of these variables don't seem necessary, as they're only used once. We should also avoid creating variables solely for renaming.